numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -30,23 +30,60 @@ class FakeShape(tuple):
     indexing, similar to the shape in CUDA Python. (Numpy shape arrays allow
     negative indexing)
     '''
+
     def __getitem__(self, k):
         if isinstance(k, int) and k < 0:
             raise IndexError('tuple index out of range')
         return super(FakeShape, self).__getitem__(k)
 
 
+class FakeWithinKernelCUDAArray(object):
+    '''
+    Created to emulate the behavior of arrays within kernels, where either array.item or array['item'] is valid.
+    This behaviour does not follow the semantics of Python and NumPy with
+    non-jitted code, and will be deprecated and removed.
```

Comment:
Who is tracking and managing this deprecation?

Reply:
I'm planning to submit a PR deprecating it, but not for 0.49

Reply:
Thanks for confirming. Is there an issue tracking it? If not please could one be created, I'll add it to the 0.50 milestone.

Reply:
(It was me that suggested that text in the docstring originally, btw)

Reply:
https://github.com/numba/numba/issues/5375